### PR TITLE
Bump Kani version to 0.34.0

### PR DESCRIPTION
**Pull request by tautschnig**
### Description of changes: 

Updated version in all `Cargo.toml` files (via
`find . -name Cargo.toml -exec sed -i 's/version = "0.33.0"/version = "0.34.0"/' {} \;`) and ran `cargo build-dev` to have `Cargo.lock` files updated.

### Resolved issues:

n/a

### Related RFC:

n/a

### Call-outs:

n/a

### Testing:

* How is this change tested? CI

* Is this a refactor change? No

### Checklist
- [x] Each commit message has a non-empty body, explaining why the change was made
- n/a Methods or procedures are documented
- [x] Regression or unit tests are included, or existing tests cover the modified code
- [x] My PR is restricted to a single feature or bugfix

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

**Review by feliperodri — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,13 +7,24 @@ This file was introduced starting Kani 0.23.0, so it only contains changes from
...
+* Support for stubbing out foreign functions ([pull request](https://github.com/model-checking/kani/pull/2658) by @feliperodri)
+* Coverage reporting without a need for cbmc-viewer ([pull request](https://github.com/model-checking/kani/pull/2609) by @adpaco-aws)
+* Add support to array-based SIMD ([pull request](https://github.com/model-checking/kani/pull/2633) by @celinval)
+* Add unchecked/SIMD bitshift checks and disable CBMC flag ([pull request](https://github.com/model-checking/kani/pull/2630) by @reisnera)
+* Bump CBMC to v5.89.0 ([pull request](https://github.com/model-checking/kani/pull/2662) by @remi-delmas-3000)
+* Update Rust toolchain to nightly 2023-08-04 ([pull request](https://github.com/model-checking/kani/pull/2661) by @remi-delmas-3000)
+
+## New Contributors
+* @reisnera made their first contribution in https://github.com/model-checking/kani/pull/2630
+
```
Comment (adpaco-aws) [inline CHANGELOG.md:27]: We don't include this section in our notes.

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,13 +7,24 @@ This file was introduced starting Kani 0.23.0, so it only contains changes from
...
 
 ## What's Changed
 
+* Allow specifying the scheduling strategy in #[kani_proof] for async functions ([pull request](https://github.com/model-checking/kani/pull/1661) by @fzaiser)
+* Support for stubbing out foreign functions ([pull request](https://github.com/model-checking/kani/pull/2658) by @feliperodri)
+* Coverage reporting without a need for cbmc-viewer ([pull request](https://github.com/model-checking/kani/pull/2609) by @adpaco-aws)
+* Add support to array-based SIMD ([pull request](https://github.com/model-checking/kani/pull/2633) by @celinval)
+* Add unchecked/SIMD bitshift checks and disable CBMC flag ([pull request](https://github.com/model-checking/kani/pull/2630) by @reisnera)
+* Bump CBMC to v5.89.0 ([pull request](https://github.com/model-checking/kani/pull/2662) by @remi-delmas-3000)
+* Update Rust toolchain to nightly 2023-08-04 ([pull request](https://github.com/model-checking/kani/pull/2661) by @remi-delmas-3000)
```
Comment (adpaco-aws) [inline CHANGELOG.md:23]: I'd rather use the default format to conform with the format in release notes for other versions.

File: CHANGELOG.md [hunk 3]
```diff
@@ -7,13 +7,24 @@ This file was introduced starting Kani 0.23.0, so it only contains changes from
 ## [0.34.0]
 
 ### Breaking Changes
-* Change default solver to CaDiCaL by @celinval in https://github.com/model-checking/kani/pull/2557
+* Change default solver to CaDiCaL ([pull request](https://github.com/model-checking/kani/pull/2557) by @celinval)
```
Comment (adpaco-aws) [inline CHANGELOG.md:10]: Same here.

File: CHANGELOG.md (hunk 2 above)
Comment (adpaco-aws) [reply CHANGELOG.md:23]: Also, isn't this missing #2663 ?

**Review by adpaco-aws — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (tautschnig) [reply CHANGELOG.md]: Reverted the beautification commit, and added #2663 back in (I wasn't sure whether that was to be considered user-facing, but the spurious verification failures that it addresses surely are).

File: CHANGELOG.md (hunk 3 above)
Comment (tautschnig) [reply CHANGELOG.md]: Done.

**Pull request merged**